hw/ip/aon_timer/doc/_index.md

```diff
@@ -0,0 +1,102 @@
+---
+title: "AON Timer Technical Specification"
+---
+
+# Overview
+
+This document specifies the Always-On ("AON") Timer IP functionality.
+This module conforms to the [Comportable guideline for peripheral functionality.]({{< relref "doc/rm/comportability_specification" >}})
+See that document for an overview of how it is integrated into the top level system.
+
+## Features
+
+- Two 32-bit upcounting timers: one timer functions as a wakeup timer, one as a watchdog timer
+- The watchdog timer has two thresholds: bark (generates an interrupt) and bite (resets core)
+
+## Description
+
+### AON Wakeup timer
+
+The always-on wakeup timer operation is straightforward.
+A count starts at 0 and slowly ticks upwards.
+When it reaches / exceeds the wake count, a level wakeup signal is sent to the power manager and a level IRQ is generated to the processor.
+This wakeup signal stays high until it is explicitly acknowledged by software.
+The wakeup timer can be used like a real-time clock for long periods of low-power mode (though it does not give any guarantees of time-accuracy TODO: specify accuracy).
+
+### AON Watchdog timer
+
+The always-on watchdog timer behaves similarly to the wakeup timer.
+It has an independent count starting at 0 which slowly ticks upwards.
+When the first threshold is met or exceeded, a level wakeup signal (if enabled) is generated to the power manager.
+Simultaneously, a level IRQ signal is also generated to the processor.
+
+If the system is in a low power state, the wakeup signal asks the power manager to wake the system such that the IRQ can be serviced.
+If the system is not in low power mode, the IRQ is immediately serviced.
+Both the wakeup and the IRQ signals remain asserted until system reset or explicit acknowledgement by software.
+This first threshold is known as the watchdog bark.
+
+When the second threshold is met (this is known as the watchdog bite), a reset request is sent to the power manager which will reset the whole system.
```

Comment:
Is the reset request guaranteed to succeed? Maybe it is worth word it more strongly, something like:
```
When the second threshold is met (this is known as the watchdog bite), a reset request is sent to the power manager which is guaranteed to trigger a System Reset.
```
Not sure what is the best way of wording it, but to me this sentence doesn't sound like the reset request is always guaranteed to succeed, even though it has `will` in the sentence.

Also, it seems that `reset the whole system` is maybe not a correct term to use here, as later on you mention the system reset - which is not whole system (POR would be whole system)?

Reply:
Agree `whole system` doesn't make sense, will change to system reset. Not sure I want to say it's guaranteed, since it's just a request (which the power manager theoretically could ignore).

Reply:
> Not sure I want to say it's guaranteed, since it's just a request (which the power manager theoretically could ignore).

Makes sense, do you know a potential use-case when such request may get ignored by the Power Manager? Seems slightly counter-intuitive if watchdog event is dropped.

Reply:
I guess it's possible that it would be ignored if the lifecycle escalation has put the system in a killed state (forcing the user to do the reset)